python/src/scipp/plot/model.py

```diff
@@ -193,87 +191,47 @@ def get_data_coord(self, name, dim):
-                                           rebin_edges[dim][dim, -1])
-            dslice = dslice[this_slice]
-
-        # Rebin the data
-        for dim, edges in rebin_edges.items():
-            dslice = sc.rebin(dslice, dim, edges)
-
-        # Divide by pixel width
-        # TODO: can this loop be combined with the one above?
-        for dim, edges in rebin_edges.items():
-            div = edges[dim, 1:] - edges[dim, :-1]
-            div.unit = sc.units.one
-            dslice /= div
-
-        return dslice
-
     def rescale_to_data(self):
         """
         Get the min and max values of the currently displayed slice.
         """
         vmin = None
         vmax = None
         if self.dslice is not None:
-            vmin = sc.min(self.dslice.data).value
-            vmax = sc.max(self.dslice.data).value
+            vmin = sc.nanmin(self.dslice.data).value
+            vmax = sc.nanmax(self.dslice.data).value
         return vmin, vmax
 
-    def _slice_or_rebin(self, array, dim, dim_slice):
-        """
-        The convetion is:
-        - if the slider thickness is zero, use index-based slicing
-        - otherwise, use `rebin`.
-        """
-        deltax = dim_slice["thickness"]
-        if deltax == 0.0:
-            array = array[dim, dim_slice["index"]]
-        else:
-            loc = dim_slice["bin_centre"]
-            array = self.resample_data(
-                array,
-                rebin_edges={
-                    dim:
-                    sc.Variable(
-                        [dim],
-                        values=[loc - 0.5 * deltax, loc + 0.5 * deltax],
-                        unit=array.coords[dim].unit)
-                })[dim, 0]
-            array *= (deltax * sc.units.one)
-        return array
-
     def slice_data(self, array, slices):
         """
         Slice the data array according to the dimensions and extents listed
         in slices.
         """
-        data_slice = array
-        for dim in slices:
-            data_slice = self._slice_or_rebin(data_slice, dim, slices[dim])
-        return data_slice
+        for dim, [lower, upper] in slices.items():
+            aslice = array[dim, lower:upper]
+            # Summing the DataArray applies the masks. To preserve the masks,
+            # we sum the data and ANY the masks
+            array = sc.DataArray(data=sc.sum(aslice.data, dim))
+            for dim_, coord in aslice.coords.items():
+                if dim_ != dim:
+                    if dim in coord.dims:
+                        # We need to slice the dim if it is still present in
+                        # the coord
+                        array.coords[dim_] = coord[dim, 0]
+                    else:
+                        array.coords[dim_] = coord
+            for m, msk in aslice.masks.items():
+                if dim in msk.dims:
+                    array.masks[m] = sc.any(msk, dim)
+                else:
+                    array.masks[m] = msk
```

Comment:
This block of code is not super pretty. Can we do things in a simpler/more efficient way?

Reply:
In the end I think using `rebin` gives better performance than this ugly piece of code..